cocotb/share/lib/vhpi/VhpiCbHdl.cpp

```diff
@@ -548,9 +549,15 @@ int VhpiSignalObjHdl::set_signal_value(long value)
     switch (m_value.format) {
         case vhpiEnumVecVal:
         case vhpiLogicVecVal: {
+            using EnumLimits = std::numeric_limits<vhpiEnumT>;
+            if ((value > EnumLimits::max()) || (value < EnumLimits::min())) {
+                LOG_ERROR("Data loss detected");
+                return -1;
+            }
             int i;
+            vhpiEnumT value_as_enum = static_cast<vhpiEnumT>(value);
```

Comment:
This cast is probably wrong - since each bit is being  treated as a separate enum value, it's not correct to assume that all the bits fit in one.